src/utils/react.js

```diff
@@ -0,0 +1,23 @@
+import { useState, useEffect, useRef } from 'react'
+
+export let useStateLens = value => {
```

Comment:
if we're gonna do this, let's make sure it's exported properly from the index as well and just make this a thing

Reply:
But I still think we should just make popover/modal/whatever not take an actual lens as part of the API